Week-4/getNumber.js

```diff
@@ -0,0 +1,110 @@
+// custom Promise function without external libraries
+[FULFILLED , REJECTED , PENDING] = ['FULFILLED', 'REJECTED', 'PENDING']; 
+/* class for myPromis */
+class myPromise {
+  /* Defining the states */ 
+  state;
+  isResolved = false;
+  isRejected = false;
+  resolvedData;
+  rejectedData;
+  thenChain = [];  //this is for the then chain
+  catchChain = []; //this is for the catch chain
+
+  constructor(exe) {
+    //Default status is pending as this promise is just created.
+    this.state = PENDING;
+
+    //resolve with Value
+    const resolve = (value) => {
+      this.state = FULFILLED;
+      this.isResolved = true;
+      this.resolvedData = value;
+      // checking the length of thenChain anything greater than zero will
+      // will call the subsequent chain, using the accumulator and the function
+      if (this.thenChain.length) {
```

Comment:
@rajeshvlega9 i don't think we need this implementation, because with the help of resolve / reject methods we only resolve or reject a value and we wont call a then or catch on that

We call then and catch on the returned promise. correct me if i am wrong

But well done, others looks good.